test/common/http/utility_test.cc

```diff
@@ -124,6 +124,18 @@ TEST(HttpUtility, TestParseCookie) {
   EXPECT_EQ(value, "abc123");
 }
 
+TEST(HttpUtility, TestParseCookieBadValues) {
+  TestHeaderMapImpl headers{{"cookie", "token1=abc123; = "},
+                            {"cookie", "token2=abc123;   "},
+                            {"cookie", "; token3=abc123;"},
+                            {"cookie", "=; token4=\"abc123\""}};
+
+  EXPECT_EQ(Utility::parseCookieValue(headers, "token1"), "abc123");
```

Comment:
Confirmed with local run that removing the `if` block that continues will result in test failure with same `"basic_string::substr", __pos=18446744073709551615` issue